Code/OpenRealEstate.Core/Guard.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+
+namespace OpenRealEstate.Core
+{
+    public class Guard

```

Comment:
Can't make it internal, this is in the `Core` proj, wouldn't not be accessible in `Services` proj  (presumably we'd further roll this out for all `ArgumentNullExceptions` in other projects too, hence why I did not create the Guard class in `Services`)